src/mono/nuget/Microsoft.NET.Runtime.MonoTargets.Sdk/Sdk/component-manifest.targets

```diff
@@ -0,0 +1,94 @@
+         in _MonoComponent -->
+    <!-- Output: _MonoRuntimeSelectedStubComponents (static linking only) subset of
+         _MonoRuntimeComponentName items that were not in _MonoComponent -->
+    <!-- Output: _MonoRuntimeUnSelectedComponents subset of _MonoRuntimeComponentName items that were not in _MonoComponent. That is, the components that are available but that weren't selected. -->
+    <!-- Output: _MonoRuntimeUnSelectedStubComponents (static linking only) subset of _MonoRuntimeComponentName items that were in _MonoComponent.  That is, components that were selected - so their stubs should not be linked. -->
+    <!-- Output: _MonoRuntimeComponentLink item list of the filenames to link (relative paths
+         relative to the native/ directory of the runtime pack). For dynamic linking, this is only
+         the enabled components.  For static linking, it also includes stubs for the non-selected
+         components -->
+    <!-- Output: _MonoRuntimeComponentDontLink item list of the filename to exclude from linking
+         (relative paths relative to the native/ directory of the runtime pack). For static linking,
+         this is stubs or components that must not be passed to the native linker. For dynamic
+         linking, this is copmonent shared libraries that must not be bundled with the app
+         -->
+    <!-- Generally the workload should only use one of _MonoRuntimeComponentLink or
+         _MonoRuntimeComponentDontLink.  For example, a workload that doesn't understand the
+         contents of native/ in details will generally bundle every native library from native/.
+         Such a workload should then use DontLink to exclude the disabled components.  On the other
+         hand a workload that explicitly pics the native libraries from native/ should use
+         _MonoRuntimeComponentLink to pick just the needed runtime component artifacts. -->
+   
+    <!-- The target also depends on the following items that are provided by the component-manifest.props from the runtime pack -->
+    <!-- _MonoRuntimeComponentLinking item list: either 'dynamic' or 'static' (defined by the runtime for each possible RuntimeIdentifier) -->
+    <!-- _MonoRuntimeComponentName item list: the components for the current RuntimeIdentifier (defined by _MonoComputeAvailableComponentDefinitions above ) -->
+
+    <ItemGroup Condition="'$(_MonoRuntimeComponentCurrentLinking)' == 'dynamic' or '$(_MonoRuntimeComponentCurrentLinking)' == 'static'">
+      <!-- take the intersection of the two item lists. -->
+      <!-- n.b. copies the metadata from _MonoRuntimeComponentName to _MonoRuntimeSelectedComponents -->
+      <_MonoRuntimeComponentNameForRid Include="@(_MonoRuntimeComponentName)" Condition="'%(RuntimeIdentifier)' == '$(RuntimeIdentifier)'" />
+      <_MonoRuntimeSelectedComponents Include="@(_MonoRuntimeComponentNameForRid)" Condition="'@(_MonoRuntimeComponentNameForRid)' == '@(_MonoComponent)' and '%(Identity)' != ''" />
```

Comment:
I *think*, `'@(_MonoRuntimeComponentNameForRid)' == '@(_MonoComponent)'` would compare the stringified version of the two items, which would be order sensitive.

Reply:
`Condition="'@(A)' == '@(B)' and '%(Identity)' != ''"` ends up comparing every item in list A to every item in list B one by one.  The `%(Identity)` makes it go item by item.  (This is apparently a trick for doing set intersections in msbuild https://philcupp.wordpress.com/2009/10/12/msbuild-tricks-i-don%E2%80%99t-want-to-forget/)